docs/recipes/authentication.md

```diff
@@ -38,9 +38,9 @@ export class AuthState {
   constructor(private authService: AuthService) {}
 
   @Action(Login)
-  login({ patchState }: StateContext<AuthStateModel>, { payload: { username } }: Login) {
+  login({ patchState }: StateContext<AuthStateModel>, { payload }: Login) {
```

Comment:
Rewrite to

{ payload: username }: Login

patchState({ token, username })

Reply:
@splincode `username` is a property of the payload so the correct deconstruction would be `{ payload: { username } }` to just get the `username` prop from the action. That being said I find that most developers get confused by advanced deconstruction so it would not be best to be in the docs as @MrOutput proposes I would leave this line as is and correct the usage at the `patchState`.

Reply:
@splincode the `username` property is contained within the payload so your proposed change would not work. It would result in something like this:
{ payload: username }: Login

patchState({ token, username.username })